src/main/java/xyz/groundx/caver_ext_kas/kas/wallet/Wallet.java

```diff
@@ -98,6 +112,84 @@ public Wallet(String chainId, ApiClient walletApiClient) {
         setApiClient(walletApiClient);
     }
 
+    /**
+     * Migrates Klaytn accounts to KAS Wallet API. <br>
+     * This method needs a RPC instance directing endpoint url of Node API. <br>
+     * Therefore, you must initialize the node api before using this function. <br>
+     *
+     * <pre>Example :
+     * {@code
+     * ArrayList<MigrationAccount> accountsToBeMigrated = new ArrayList<>();
+     * accountsToBeMigrated.add(new MigrationAccount("0x{address}", "0x{privateKey}"));
+     * accountsToBeMigrated.add(new MigrationAccount("0x{address}", "0x{privateKey}", "0x{nonce}"));
+     *
+     * RegistrationStatusResponse response = caver.kas.wallet.migrateAccounts(accountsToBeMigrated);
+     * }</pre>
+     *
+     * @param accounts A list of accounts to be migrated to KAS Wallet Service.
+     * @return RegistrationStatusResponse
+     * @throws ApiException
+     * @throws IOException
+     * @throws NoSuchFieldException
+     */
+    public RegistrationStatusResponse migrateAccounts(List<MigrationAccount> accounts) throws ApiException, IOException, IllegalArgumentException, NoSuchFieldException {
+        if (this.rpc == null) {
+            throw new NoSuchFieldException("Before using migrateAccounts, initNodeAPI must  be called first.");
+        }
+
+        if (this.rpc.getWeb3jService() instanceof HttpService) {
+            String url = ((HttpService) this.rpc.getWeb3jService()).getUrl();
+            if (!url.contains("klaytnapi")) {
+                System.out.println("WARN: Endpoint URL of Node API is " + url + " which is not klaytnapi.");
+                System.out.println("WARN: You should initialize Node API with working endpoint url before calling migrateAccounts.");
+            }
+        }
+
+        // Need to validate whether given list of migration accounts is valid or not.
+        for (int i=0; i<accounts.size(); i++) {
+            if (validateMigrationAccount(accounts.get(i)) == false) {
+                throw new IllegalArgumentException("Given MigrationAccount is not valid.");
+            };
+        }
+
+        AccountRegistrationRequest request = new AccountRegistrationRequest();
+
+        KeyCreationResponse keyCreationResponse = this.createKeys(accounts.size());
+        List<Key> createdKeys = keyCreationResponse.getItems();
+
+        String gasPrice = this.rpc.klay.getGasPrice().send().getResult();
```

Comment:
Do you think this should be existed in this logic still?

Reply:
@jimni1222 
Actually gasPrice is fetched in `AbstractTransaction.fillTransaction` but If we fetch it first, we can decrease RPC calls fetching gasPrice for each account.

Reply:
Oh, okay. We can get benefit from not doing that for every account. Sounds good :)